core/src/eval/contract_eq.rs

```diff
@@ -161,8 +161,8 @@ fn contract_eq_bounded(
                     let had_gas = state.use_gas();
                     state.use_gas();
 
-                    let closure1 = idx1.borrow();
-                    let closure2 = idx2.borrow();
+                    let closure1 = idx1.borrow_orig();
```

Comment:
This was lost in the original extraction of `contract_eq`. I notice it thanks to a unused warning popping up after getting rid of the legacy typecheck module.

Reply:
Is it worth having a test that would have caught this?